tests/worktree/worktree.c

```diff
@@ -228,6 +228,30 @@ void test_worktree_worktree__init(void)
 	git_repository_free(repo);
 }
 
+void test_worktree_worktree__add_from_bare(void)
+{
+	git_worktree *wt;
+	git_repository *repo, *wtrepo;
+	git_buf path = GIT_BUF_INIT;
+
+	repo = cl_git_sandbox_init("testrepo.git");
+
+	cl_assert_equal_i(1, git_repository_is_bare(repo));
+	cl_assert_equal_i(0, git_repository_is_worktree(repo));
+
+	cl_git_pass(git_buf_joinpath(&path, repo->gitdir, "../worktree-frombare"));
+	cl_git_pass(git_worktree_add(&wt, repo, "worktree-frombare", path.ptr, NULL));
+
+	cl_git_pass(git_repository_open(&wtrepo, path.ptr));
+	cl_assert_equal_i(1, git_repository_is_bare(wtrepo));
+	cl_assert_equal_i(1, git_repository_is_worktree(wtrepo));
+
```

Comment:
`git_repository__ensure_not_bare` should be tested again on `repo` to make sure that adding a worktree to a bare repository does not change the return value on the bare repository.

Reply:
Agreed

Reply:
Note that because of the semantic change in `git_repository_is_bare`, `git_repository__ensure_not_bare` now has the same semantics by default. So apart from the special error handling, it's equivalent (and I think it's better to test the public low-level API part vs. a slightly higher-up private one).